lib/fixture/locator.ts

```diff
@@ -0,0 +1,110 @@
+import {promises as fs} from 'fs'
+
+import type {PlaywrightTestArgs, TestFixture} from '@playwright/test'
+import {selectors} from '@playwright/test'
+
+import {queryNames} from '../common'
+import type {LocatorQueries as Queries, AllQuery, Query, SelectorEngine} from './types'
+import {replacer, reviver} from './helpers'
+
+const queriesFixture: TestFixture<Queries, PlaywrightTestArgs> = async ({page}, use) => {
+  const queries = queryNames.reduce(
+    (rest, query) => ({
+      ...rest,
+      [query]: (...args: Parameters<Queries[keyof Queries]>) =>
+        page.locator(`__testing_library__=${JSON.stringify({query, args}, replacer)}`),
```

Comment:
Hm. I guess the loss here is that this will be a bit more difficult to read in the [Playwright Inspector](https://playwright.dev/docs/inspector), however there are pros in that it gives us full usage of the Testing Library API and we're unlikely to ever clash with the Playwright `Locator` API (which I've already done with the new `role=` selector API myself.)

Reply:
> Hm. I guess the loss here is that this will be a bit more difficult to read in the [Playwright Inspector](https://playwright.dev/docs/inspector)

This is a great point. I should be testing this out with the inspector for sure. I guess my intention here was to avoid conflicts (🤔 pretty unlikely) and make it very clear that we're serializing stuff and that the selectors are not meant to be used directly. I think you're right, though. I'll play around with the inspector.

Reply:
See https://github.com/testing-library/playwright-testing-library/pull/403#issuecomment-1111367758